src/rpcclient/rpcclient/darwin/syslog.py

```diff
@@ -111,11 +124,67 @@ def get_subsystem(self, name: str) -> OsLogPreferencesSubsystem:
 class Syslog:
     """" manage syslog """
 
-    def __init__(self, client):
+    def __init__(self, client: Client):
         self._client = client
         self._load_logging_support_library()
         self.preferences_manager = OsLogPreferencesManager(self._client)
 
+    @cached_property
+    def _enable_har_global(self):
+        r"""
+        In order to find an unexported global variable, we use a sequence of instructions that we know comes after
+        the aforementioned variable is accessed.
+
+            88 07 2D D0                 ADRP            X8, #har_global@PAGE
+            08 41 55 39                 LDRB            W8, [X8,#har_global@PAGEOFF]
+            08 03 00 34                 CBZ             W8, loc_187DA1294
+            E0 03 13 AA                 MOV             X0, X19 ; __str
+            01 00 80 D2                 MOV             X1, #0  ; __endptr
+            42 01 80 52                 MOV             W2, #0xA ; __base
+
+        After getting the address of the instruction `mov x0,x19`, we need to rewind 2 instructions. Next, we verify
+        that the above 2 instructions are `adrp` && `ldrb` . Then, extract the page address + page offset and
+        calculate the affective address.
+        """
+
+        address = []
+        regex_hex = r'0x[0-9a-fA-F]+'
+
+        self._client.load_framework('CFNetwork')
+        cfnetwork_base = self._client.modules.get('/System/Library/Frameworks/CFNetwork.framework/CFNetwork')
+        pattern_addr = self._client.symbols.memmem(cfnetwork_base, 0xffffffff, PATTERN, len(PATTERN))
+        pattern_sym = self._client.symbol(pattern_addr)
+        disass = (pattern_sym - len(PATTERN)).disass(8)
+        if disass[0].mnemonic != 'adrp' and disass[1].mnemonic != 'ldrb':
+            raise HarGlobalNotFoundError()
+
+        for instruction in disass:
+            address.append(
+                int(re.findall(regex_hex, instruction.op_str)[0], 16)
+            )
+        return sum(address)
+
+    def enable_harlogger_for_process(self, pid: int) -> None:
+
+        process = self._client.processes.get_by_pid(pid)
+        self._enable_harlogger_for_process(process)
+        self.set_har_capture_global(True)
+
+    def enable_harlogger_for_all(self) -> None:
```

Comment:
add an optional `expression: str` to match the path of iterated processes. if the expression is given, then only enable for processes containing it